### ISSUE TITLE
calculation of accuracy 

### ISSUE THREAD

**Issue by bill2239** [arunpatala/captcha.irctc#20]
hi,
I would like to know if the way of calculate accuracy need to change as I change the dataset. For example, I have 9 classes for my captcha (the length of captcha is 9 instead of 5). so in the accuracy calculation.
function train.accuracy(Xv,Yv,net,batch)
    net:evaluate()
    local batch = batch or 64
    local Nv = Xv:size(1)
    local lloss = 0
    for i =1,Nv,batch do
        local j = math.min(i+batch-1,Nv)
        local Xb = Xv[{{i,j}}]:cuda()
        local Yb = Yv[{{i,j}}]:cuda()
        local out = net:forward(Xb) -- N*k*C
        local tmp,YYb = out:max(3)
        lloss = lloss + YYb:eq(Yb):sum()
    end
    return (100*lloss/(5*Nv))
end
is this line
return (100*lloss/(5*Nv)) need to change to return (100*lloss/(9*Nv))? also, what is the out:max(3) means is it a parameter need to change to adapt the dataset?.
also, is train.accuracyK calculate the accuracy if all characters are predicted correctly?
if anyone have some idea of this issue, I would be really appreciate it. thanks.
